plugins/nf-nomad/src/main/nextflow/nomad/executor/FailsafeExecutor.groovy

```diff
@@ -0,0 +1,66 @@
+package nextflow.nomad.executor
+
+import dev.failsafe.Failsafe
+import dev.failsafe.RetryPolicy
+import dev.failsafe.event.EventListener
+import dev.failsafe.event.ExecutionAttemptedEvent
+import dev.failsafe.function.CheckedSupplier
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.nomadproject.client.ApiException
+import nextflow.nomad.config.RetryConfig
+
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.TimeoutException
+import java.util.function.Predicate
+
+@Slf4j
+@CompileStatic
+class FailsafeExecutor {
+
+    private RetryConfig config
+
+    FailsafeExecutor(RetryConfig config){
+        this.config = config
+    }
+
+    protected <T> RetryPolicy<T> retryPolicy(Predicate<? extends Throwable> cond) {
+
+        final listener = new EventListener<ExecutionAttemptedEvent<T>>() {
+            @Override
+            void accept(ExecutionAttemptedEvent<T> event) throws Throwable {
+                log.debug("Nomad TooManyRequests response error - attempt: ${event.attemptCount}; reason: ${event.lastFailure.message}")
+            }
+        }
+        return RetryPolicy.<T>builder()
+                .handleIf(cond)
+                .withBackoff(config.delay.toMillis(), config.maxDelay.toMillis(), ChronoUnit.MILLIS)
+                .withMaxAttempts(config.maxAttempts)
+                .withJitter(config.jitter)
+                .onRetry(listener)
+                .build()
+    }
+
+    final private static List<Integer> RETRY_CODES = List.of(408, 429, 500, 502, 503, 504)
```

Comment:
@jagedn, is there a specific dictionary/reference you used to focus only on these exit codes? 

Maybe a good idea to add that in the comments or just explain the through process behind these exit codes.

Reply:
hahaha c&p from the azure plugin

Reply:
Ah, in that case these must be specific to the Azure Batch API 🤔 


@tomiles @matthdsm @jhaezebr , are you aware of any Nomad specific error codes or does a Nomad client/server just propogate the task-and-OS level error code?

Reply:
don't think so, they are common http error codes

I've added descriptions for them

Reply:
Mmm, I was under the impression that nomad would have it's own set of error codes as well but judging from this issue https://github.com/hashicorp/nomad/issues/17782 I think that for client/task errors it stores them in `exit code` 


Could be related to https://github.com/nextflow-io/nf-nomad/issues/77 🤔

Reply:
yes, I agree but maybe better handle the ext code in the #77 and let this PR handle the infra/http errors